### PR TITLE
Fixed section IDs and relative links to other docs

### LINKED ISSUE #553: Cleanup tasks to prepare for docs publication
Following the asciidoc conversion, there are some tasks that still need to be performed on the overall doc set in order to get it ready for publication:
- Update and formalize permalink IDs (for each h2, h3). Currently they're auto-generated in the asciidoc files, but because they will ultimately end up in the URL breadcrumbs it's probably a good idea to review them and make sure they're concise and accurate (it'll be harder to change later and if we can avoid redirects that's a good thing (tm) ).
- Cross-document (and some cross-repo) links currently refer to the adoc files. We need to replace these with the published URLs (this should probably wait until we do a test publication and have the full URL), otherwise users will end up in the GH files instead of the published content.

### PR DESCRIPTION
**Pull request by thatdocslady**
Fixes #553 

This PR addresses the following issues:

- Fix section/topic IDs after asciidoc conversion (many of them were long and clumsy)
- Fix links to other docs in the repo to be relative (instead of hard-coded) to allow them to parse into HTML links when we publish docs later
- Fixed links that were broken due to line breaks 
- Minimize links to other repos (only a handful of them are currently used and only when absolutely necessary)

Reviews welcome!

### CODE REVIEW

File: docs/installing.adoc [hunk 1]
```diff
@@ -252,6 +252,30 @@ $ curl -sL https://raw.githubusercontent.com/projectatomic/adb-atomic-developer-
 Refer:
 link:../components/centos/centos-mesos-marathon-singlenode-setup/README.adoc[README]
 
+<<<<<<< a7f6568029db13e9f431b881a8ba11942dd28eb4
```
Comment (LalatenduMohanty) [inline docs/installing.adoc]: @thatdocslady Please fix this. I think this is a byproduct of rebase.

File: docs/installing.adoc [hunk 2]
```diff
@@ -285,7 +309,7 @@ export BOX=<box-name>
 ----
 ====
 
-[[set-up-adb-behind-an-http-proxy]]
+[[setup-http-proxy]]
 == Step 6. (Optional) Set up ADB behind an HTTP proxy
```
Comment (LalatenduMohanty) [inline docs/installing.adoc:289]: `setup-http-proxy` looks incorrect. The section represents `how to setup ADB behind a proxy setup`

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

**Comment by thatdocslady** [comment]
@LalatenduMohanty fixed the sections per your feedback, thanks! @hferentschik @Preeticp can you also please review? Cheers :)

**Review by LalatenduMohanty — APPROVED** [review]

**Comment by LalatenduMohanty** [comment]
@Preeticp waiting for your review. Thanks.

**Comment by Preeticp** [comment]
There is just this one substep in step 5, in the Installation guide, being discussed, apart from that, the rest LGTM.

**Comment by thatdocslady** [comment]
@Preeticp fixed step 5 in the install doc per our discussion.

**Comment by Preeticp** [comment]
LGTM @thatdocslady, @LalatenduMohanty

**Pull request merged**